org.uqbar.project.wollok/src/org/uqbar/project/wollok/validation/WollokDslValidator.xtend

```diff
@@ -820,7 +821,11 @@ class WollokDslValidator extends AbstractConfigurableDslValidator {
 	@CheckGroup(WollokCheckGroup.POTENTIAL_PROGRAMMING_PROBLEM)
 	def nonBooleanValueInIfCondition(WIfExpression it) {
 		if (!condition.isBooleanOrUnknownType) {
-			report(WollokDslValidator_EXPECTING_BOOLEAN, it, WIF_EXPRESSION__CONDITION)
+			if (condition instanceof WAssignment) {
```

Comment:
Esto sólo chequea que la asignación esté en la raiz del nodo condición, hay otras posibilidades que no estamos considerando. 
- En este mismo punto, si hubiera un árbol de condicionales (tipo A and B or C), sería interesante hacer la misma validación sobre sus componentes.
- En otros lugares donde aparezca el = en un lugar donde se espera una expresión (como argumento en un mensaje, return, etc).

No pararía el PR por esto, pero tal vez podríamos registrar esos issues.

Reply:
Ahí le agregué validaciones a la condición binaria.
En el return el mensaje que tira es "Cannot return an assignment".
Me parece bien continuarlo como issues donde evaluemos qué tipo de mensaje salta y cómo lo queremos tunear.